pkgs/historical-modules/default.nix

```diff
@@ -104,15 +105,24 @@ let
     }
   ];
 
-  moduleFromHistory = { moduleId, commit, deployment ? false }:
+  moduleFromHistory = { displayVersion, moduleId, commit, deployment ? false }:
     let
       flake = getFlake "github:replit/nixmodules/${commit}";
+      module =
+        if deployment then
+          (flake.deploymentModules or flake.modules).${moduleId}
+        else
+          flake.modules.${moduleId};
+      # Sneaky way of adding a displayVersion to a historical module:
+      # 1. build the output JSON file
+      # 2. parse the JSON file
+      # 3. add the displayVersion field
+      # 4. return a new derivation that builds the modified JSON contents
+      moduleJSON = fromJSON (builtins.unsafeDiscardStringContext (readFile module.outPath)) // {
```

Comment:
This is pretty scary. Are you sure it still includes the right files in the historical module closure when you do this?

Do we need it in the actual module json, or just the registry?

Reply:
Okay, can you not discard the string context? I think we need it.

Reply:
It may be simpler to do this as derivation.

Reply:
I guess I need to somehow parse the JSON without stripping the closure context, but in a way it will let me do it.

Reply:
Does
```
fromJSON ((readFile module.outPath) // {
...
```
not do that?

Reply:
with no context, attempting to try it yielded:
<img width="1510" alt="image" src="https://github.com/replit/nixmodules/assets/278900/9d05ab3f-2fc8-4e3e-9bf5-57dd2a8252b3">

Reply:
Okay, I think this works: https://github.com/replit/nixmodules/pull/344/files

Reply:
Will take your change as it has a existence check that mine doesn't.